### PR TITLE
api: manifest based edge default documentation.

### PR DESCRIPTION
**Pull request by htuch**
This PR replaces #11058, taking a slightly different approach. We utilize field options to annotate fields that should be set for untrusted environments with [configure_for_untrusted_downstream, configure_for_untrusted_downstream]. Defaults are provided out-of-band, in a manifest files in docs/edge_defaults_manifest.yaml.
    
Protodoc glues the manifest and options together when generating field documentation, providing an additional notice for sensitive fields.
    
Risk level: Low (docs only).
Testing: Inspection of generated docs.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: api/envoy/config/bootstrap/v4alpha/bootstrap.proto [hunk 1]
```diff
@@ -137,7 +138,10 @@ message Bootstrap {
   Admin admin = 12;
 
   // Optional overload manager configuration.
-  overload.v3.OverloadManager overload_manager = 15;
+  overload.v3.OverloadManager overload_manager = 15 [
```
Comment (htuch) [inline api/envoy/config/bootstrap/v4alpha/bootstrap.proto:141]: The plan is to fill out these annotations and the manifest examples for the remaining default edge examples if we agree on this approach.

**Comment by htuch** [comment]
This is now ready for general review, as the base #11108 has merged.

File: tools/config_validation/validate_fragment.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# Validate a YAML fragment against an Envoy API proto3 type.
```
Comment (mattklein123) [inline tools/config_validation/validate_fragment.py:1]: This is so, so, so close to the sphinx plugin. ;)

File: tools/protodoc/protodoc.py [hunk 3]
```diff
@@ -388,7 +401,30 @@ def FormatAnchor(label):
   return '.. _%s:\n\n' % label
 
 
-def FormatFieldAsDefinitionListItem(outer_type_context, type_context, field):
+def FormatSecurityOptions(security_option, field, type_context, edge_default_yaml):
+  sections = []
+
+  if security_option.configure_for_untrusted_downstream:
+    sections.append(
+        Indent(4, 'This field should be configured in the presence of untrusted *downstreams*'))
```
Comment (mattklein123) [inline tools/protodoc/protodoc.py:409]: nit: end with period, same below (noticed looking at rendered docs).

File: tools/protodoc/protodoc.py [hunk 4]
```diff
@@ -388,7 +401,30 @@ def FormatAnchor(label):
...
+  validate_fragment.ValidateFragment(field.type_name[1:], edge_default_yaml)
+  field_name = type_context.name.split('.')[-1]
+  example = {field_name: edge_default_yaml}
+  sections.append(
+      Indent(4, 'Example configuration for untrusted environments:\n\n') +
+      Indent(4, '.. code-block:: yaml\n\n') +
+      '\n'.join(IndentLines(6,
+                            yaml.dump(example).split('\n'))))
+
+  return '.. note::\n' + '\n\n'.join(sections)
```
Comment (mattklein123) [inline tools/protodoc/protodoc.py:423]: Thoughts on attention vs. note? I could go either way but security seems like an attention type of thing.

File: tools/config_validation/validate_fragment.py (hunk 2 above)
Comment (htuch) [reply tools/config_validation/validate_fragment.py:1]: Right, @dmitri-d is working on the Sphinx RST plugin side of this at the moment.

**Pull request merged**